src/Congo.RazorPages/Pages/Products.cshtml.cs

```diff
@@ -18,9 +19,11 @@ public ProductsModel(IProductsService productsService)
             _productsService = productsService;
         }
 
-        public void OnGet()
+        public async Task<IActionResult> OnGet()
         {
-            SampleData = _productsService.GetSampleProducts();
+            SampleData = await _productsService.GetProducts();
+
+            return Page();
         }
 
         public ActionResult OnPostPurchase(int id)
```

Comment:
You have an error here.
This needs to be updated to call your new Purchase method.
Also please update the message to display the order id of our purchase.